openmc/cell.py

```diff
@@ -79,13 +79,12 @@ class Cell(object):
     translation : Iterable of float
         If the cell is filled with a universe, this array specifies a vector
         that is used to translate (shift) the universe.
-    offsets : ndarray
-        Array of offsets used for distributed cell searches
-    distribcell_index : int
-        Index of this cell in distribcell arrays
-    distribcell_paths : list of str
-        The paths traversed through the CSG tree to reach each distribcell
-        instance
+    paths : list of str
+        The paths traversed through the CSG tree to reach each cell
+        instance. This property is initialized by calling the
+        :meth:`Geometry.determine_paths` method.
```

Comment:
Yeah I really like your first proposal  - automatic path tracing when linking with a summary if a distribcell filter is in a tally. Might this also work if a `Cell` is found with a `distribmat` fill?

In general I'm not necessarily too keen on having backward links in the CSG tree. If this simply meant that each primitive has a `_root` attribute then perhaps this would be fine (albeit a bit magical). But I wouldn't vote to connect each primitive to the level above in the CSG tree - *i.e.*, a `Cell` pointing to its containing `Universe` - since I still think it would be beneficial to allow cells to be used in more than a single universe one day (see convo on #458`).